sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -657,6 +657,60 @@ pi_result piProgramCreateWithBinary(
   return ret_err;
 }
 
+std::vector<std::string> split_string(const std::string &str, char delimeter) {
+  std::vector<std::string> result;
+  size_t beg = 0;
+  size_t length = 0;
+  for (const auto &x : str) {
+    if (x == delimeter) {
+      result.push_back(str.substr(beg, length));
+      beg += length + 1;
+      length = 0;
+      continue;
+    }
+    length++;
+  }
+  if (length != 0) {
+    result.push_back(str.substr(beg, length));
+  }
+  return result;
+}
+
+pi_result piProgramHasKernel(pi_program program, const char *kernel_name,
+                             bool *has_kernel) {
+  assert(has_kernel != nullptr);
+
+  size_t Size;
+  cl_int Res =
+      clGetProgramInfo(cast<cl_program>(program), PI_PROGRAM_INFO_KERNEL_NAMES,
+                       0, nullptr, &Size);
+  if (Res != CL_SUCCESS) {
+    *has_kernel = false;
+    return cast<pi_result>(Res);
+  }
+
+  std::string ClResult(Size, ' ');
+  Res =
+      clGetProgramInfo(cast<cl_program>(program), PI_PROGRAM_INFO_KERNEL_NAMES,
+                       ClResult.size(), &ClResult[0], nullptr);
+  if (Res != CL_SUCCESS) {
+    *has_kernel = false;
+    return cast<pi_result>(Res);
+  }
+
+  // Get rid of the null terminator
+  ClResult.pop_back();
+  std::vector<std::string> KernelNames(split_string(ClResult, ';'));
+  for (const auto &Name : KernelNames) {
+    if (Name == kernel_name) {
```

Comment:
It seems like we do more work here than we need to. First we split `ClResult` into its constituents and then we look for one specific name inside that list. It would make sense to change `split_string` into a function that checks for a given string between the delimiters, rather than saving every string.

Reply:
Sounds good. The implementation comes from the old `has_cl_kernel` function. I agree that the approach should be changed. Ill update the function.